internal/scheduler/legacyscheduler.go

```diff
@@ -691,6 +724,32 @@ func (sched *LegacyScheduler) Schedule(
 	return jobs, nil
 }
 
+func GangIdAndCardinalityFromJob(job *api.Job, gangIdAnnotation, gangCardinalityAnnotation string) (string, int, bool, error) {
+	return GangIdAndCardinalityFromAnnotations(job.Annotations, gangIdAnnotation, gangCardinalityAnnotation)
+}
+
+func GangIdAndCardinalityFromAnnotations(annotations map[string]string, gangIdAnnotation, gangCardinalityAnnotation string) (string, int, bool, error) {
+	if annotations == nil {
+		return "", 0, false, nil
+	}
+	gangId, ok := annotations[gangIdAnnotation]
+	if !ok {
+		return "", 0, false, nil
+	}
+	gangCardinalityString, ok := annotations[gangCardinalityAnnotation]
+	if !ok {
+		return "", 0, false, errors.Errorf("missing annotation %s", gangCardinalityAnnotation)
+	}
+	gangCardinality, err := strconv.Atoi(gangCardinalityString)
+	if err != nil {
+		return "", 0, false, errors.WithStack(err)
+	}
+	if gangCardinality == 0 {
```

Comment:
should this be `if gangCardinality <= 0`

Reply:
Yes :)